### PR TITLE
Progressive override: fix query planner cache warmup

### PR DESCRIPTION
**Pull request by Geal**
The query planner needs various metadata along with the query to influence planning, and due to historical reasons that data was passed through the request context, not the planner request. The progressive override labels list was communicated like that, but missed the case of query plan cache warmup, where upon schema or configuration updates, the router takes the list of most used queries from the query planner's in memory cache, and plans them again before activating the new schema or configuration. Due to the mistake, the labels were not transmitted during warmup, which resulted in queries correctly using the overridden fields, but after an update, would revert to non overridden fields, and could not recover (unless the plan was evicted from the cache).

This fices the cache warmup issue, and there is a follow up PR to prevent further issues of that kind: https://github.com/apollographql/router/pull/6208

---

**Checklist**

Complete the checklist (and note appropriate exceptions) before the PR is marked ready-for-review.

- [ ] Changes are compatible[^1]
- [ ] Documentation[^2] completed
- [ ] Performance impact assessed and acceptable
- Tests added and passing[^3]
    - [ ] Unit Tests
    - [ ] Integration Tests
    - [ ] Manual Tests

**Exceptions**

*Note any exceptions here*

**Notes**

[^1]: It may be appropriate to bring upcoming changes to the attention of other (impacted) groups. Please endeavour to do this before seeking PR approval. The mechanism for doing this will vary considerably, so use your judgement as to how and when to do this.
[^2]: Configuration is an important part of many changes. Where applicable please try to document configuration examples.
[^3]: Tick whichever testing boxes are applicable. If you are adding Manual Tests, please document the manual testing (extensively) in the Exceptions.

### CODE REVIEW

**Comment by Geal** [comment]
I think I got confirmation that our docs are wrong: if we want to override labels, we need to do that at the router stage, not the supergraph stage, because rhai, coprocessor and native plugins execute after internal plugins:

```
024-10-02T16:56:48.182117Z INFO supergraph{graphql.operation.name=progressive3,graphql.document=query progressive3 { percent100 { foo } },graphql.operation.name=progressive3,} router{http.method=POST,http.request.method=POST,http.route=/,http.flavor=HTTP/1.1,trace_id=d79ed1c5bac19f367c7ba33d6d077851,url.path=/,client.name=custom_name,client.version=1.0,} request{http.method=POST,http.request.method=POST,http.route=/,http.flavor=HTTP/1.1,}  label percent(100) random percentage: 84.9674671949615 compared to 100
2024-10-02T16:56:48.182144Z INFO supergraph{graphql.operation.name=progressive3,graphql.document=query progressive3 { percent100 { foo } },graphql.operation.name=progressive3,} router{http.method=POST,http.request.method=POST,http.route=/,http.flavor=HTTP/1.1,trace_id=d79ed1c5bac19f367c7ba33d6d077851,url.path=/,client.name=custom_name,client.version=1.0,} request{http.method=POST,http.request.method=POST,http.route=/,http.flavor=HTTP/1.1,}  ProgressiveOverridePlugin: overridden labels: ["percent(100)"]
2024-10-02T16:56:48.182224Z INFO rhai_plugin{rhai service=supergraph :: Request,} supergraph{graphql.operation.name=progressive3,graphql.document=query progressive3 { percent100 { foo } },graphql.operation.name=progressive3,} router{http.method=POST,http.request.method=POST,http.route=/,http.flavor=HTTP/1.1,trace_id=d79ed1c5bac19f367c7ba33d6d077851,url.path=/,client.name=custom_name,client.version=1.0,} request{http.method=POST,http.request.method=POST,http.route=/,http.flavor=HTTP/1.1,}  supergraph (supergraph) rhai called
2024-10-02T16:56:48.182545Z INFO rhai_plugin{rhai service=supergraph :: Request,} supergraph{graphql.operation.name=progressive3,graphql.document=query progressive3 { percent100 { foo } },graphql.operation.name=progressive3,} router{http.method=POST,http.request.method=POST,http.route=/,http.flavor=HTTP/1.1,trace_id=d79ed1c5bac19f367c7ba33d6d077851,url.path=/,client.name=custom_name,client.version=1.0,} request{http.method=POST,http.request.method=POST,http.route=/,http.flavor=HTTP/1.1,}  unresolved: ["foo", "bar", "baz"] target=tests/samples/enterprise/progressive-override/rhai/main.rhai
2024-10-02T16:56:48.182597Z INFO rhai_plugin{rhai service=supergraph :: Request,} supergraph{graphql.operation.name=progressive3,graphql.document=query progressive3 { percent100 { foo } },graphql.operation.name=progressive3,} router{http.method=POST,http.request.method=POST,http.route=/,http.flavor=HTTP/1.1,trace_id=d79ed1c5bac19f367c7ba33d6d077851,url.path=/,client.name=custom_name,client.version=1.0,} request{http.method=POST,http.request.method=POST,http.route=/,http.flavor=HTTP/1.1,}  override: ["percent(100)"] target=tests/samples/enterprise/progressive-override/rhai/main.rhai
```

**Comment by Geal** [comment]
rebased over main to ease up the release

**Review by bnjjj — APPROVED** [review]
Missing changelog but LGTM

File: .circleci/config.yml [hunk 1]
```diff
@@ -1,5 +1,7 @@
 version: 2.1
 
+# Cache key bump: 1
```
Comment (goto-bus-stop) [inline .circleci/config.yml:3]: can we remove this now?

**Review by goto-bus-stop — APPROVED** [review]

File: .circleci/config.yml (hunk 1 above)
Comment (abernix) [reply .circleci/config.yml:3]: Yes.  We can also leave it and just change it to `2` next time we need it.  lol.  (I do usually remove it.)

**Pull request merged**